### PR TITLE
Add GET providers endpoint to Incremental Entity Provider admin routes

### PR DESCRIPTION
**Pull request by taras**
## Hey, I just made a Pull Request!

I'm building UI that can be used to manage Incremental Entity Provides. I need to get a list of available incremental entity providers that I can operate on. 

I made 2 changes
1. Removed paths file and moves all of the paths directly into the router because it was difficult to see what the route was.
2. Added GET `/incremental/providers` endpoint

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-incremental-ingestion/src/router/routes.ts [hunk 1]
```diff
@@ -50,13 +49,13 @@ export class IncrementalProviderRouter {
     });
 
     // Clean up and pause all providers
-    router.post(PROVIDER_CLEANUP, async (_, res) => {
+    router.post('/incremental/health', async (_, res) => {
```
Comment (freben) [inline plugins/catalog-backend-module-incremental-ingestion/src/router/routes.ts:52]:
```suggestion
    router.post('/incremental/cleanup', async (_, res) => {
```

File: plugins/catalog-backend-module-incremental-ingestion/src/router/routes.ts [hunk 2]
```diff
@@ -151,8 +153,17 @@ export class IncrementalProviderRouter {
       }
     });
 
+    router.get(`/incremental/providers/`, async (_req, res) => {
```
Comment (freben) [inline plugins/catalog-backend-module-incremental-ingestion/src/router/routes.ts:156]:
```suggestion
    router.get(`/incremental/providers`, async (_req, res) => {
```

File: plugins/catalog-backend-module-incremental-ingestion/src/router/routes.ts (hunk 1 above)
Comment (taras) [reply plugins/catalog-backend-module-incremental-ingestion/src/router/routes.ts]: Oops. I had a commit in my local that I forgot to push. Good catch.

**Review by freben — APPROVED** [review]
Looks good! I'll just go ahead and assume that e2e isn't affected by this.

**Pull request merged**